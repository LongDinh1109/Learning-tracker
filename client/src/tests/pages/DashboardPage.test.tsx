import { render, screen } from '@testing-library/react';
import Dashboard from '@/pages/Dashboard';

describe('Dashboard Component', () => {
  it('renders dashboard text', () => {
    render(<Dashboard />);
    const dashboardElement = screen.getByText(/dashboard/i);
    expect(dashboardElement).toBeInTheDocument();
  });

  it('renders with correct element type', () => {
    const { container } = render(<Dashboard />);
    const divElement = container.querySelector('div');
    expect(divElement).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<Dashboard />);
    expect(container).toMatchSnapshot();
  });
});
