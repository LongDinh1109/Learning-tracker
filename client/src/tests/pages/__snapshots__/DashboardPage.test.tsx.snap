// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component > matches snapshot 1`] = `
<div>
  <div>
    Dashboard
  </div>
</div>
`;

exports[`Dashboard Component matches snapshot 1`] = `
<div>
  <div>
    Dashboard
  </div>
</div>
`;
